### PR TITLE
Issue1476: cypress tests for vocab home page tempate statistics

### LINKED ISSUE #1476: Cypress tests for vocab-home page tempate (statistics)
## Description of the enhancement
The statistics on the front page of the vocabulary need to be tested. We need to test and check that the values in the Resource counts by type section are not empty or zero. The values in the Term counts by language section may be zeroes, but not blanks. Work has started and progress can be followed here:
https://github.com/NatLibFi/Skosmos/blob/cypress-tests-vocab-home-statistics/cypress/e2e/vocab-home.cy.js

## Who are the users that would benefit from the enhancement and how?
All Skosmos 3 users

(Name specific organizations, roles and/or communities if possible, and/or point to relevant discussion elsewhere)

## What new functionalities would the enhancement make possible?
Tests contribute to a better presentation of statistics

## Why is the enhancement important?
- Skosmos is coded according to the Test Driven Development principle. 
- Prevents a developer from coding an implementation that does not work

(Why should this be prioritized, instead of spending energy on something else? Who would do the work to make this happen?)

### PR DESCRIPTION
**Pull request by miguelvaara**
## Reasons for creating this PR
The tests for the vocabulary homepage contribute to providing a positive user experience. Run according to the standard Cypress testing procedure.

## Link to relevant issue(s), if any
- Closes #1476

## Description of the changes in this PR
In addition to the previously written tests, we are now testing both tables on the vocabulary homepage:
"Resource counts by type"
"Term counts by language"

## Known problems or uncertainties in this PR
Due to the issues mentioned in [this issue](https://github.com/NatLibFi/Skosmos/issues/1561), the previously created tests regarding the alphabetical list are failing inconsistently

## Checklist

- [ ] phpUnit tests pass locally with my changes
- [ ] I have added tests that show that the new code works, or tests are not relevant for this PR (e.g. only HTML/CSS changes)
- [x] The PR doesn't reduce accessibility of the front-end code (e.g. tab focus, scaling to different resolutions, use of `.sr-only` class, color contrast)
- [x] The PR doesn't introduce unintended code changes (e.g. empty lines or useless reindentation)

### CODE REVIEW

File: tests/cypress/template/vocab-home.cy.js [hunk 1]
```diff
@@ -1,42 +1,134 @@
...
+    // Check that the first entry is "abstract objects"
+    entries.first().invoke('text').should('equal', 'abstract objects')
   })
   it('alphabetical index letters are clickable', () => {
-    cy.visit('/test/en') // go to the "Test ontology" home page
+    cy.visit('/yso/en') // Go to the "YSO - General Finnish ontology (archaeology)" home page
 
-    // click on the second letter (C)
-    cy.get('#tab-alphabetical .pagination :nth-child(2) > .page-link').click()
+    // Click on the second letter (D)
```
Comment (osma) [inline tests/cypress/template/vocab-home.cy.js:33]: Shouldn't this be "fourth letter"?

File: tests/cypress/template/vocab-home.cy.js [hunk 2]
```diff
@@ -1,42 +1,134 @@
 describe('Vocabulary home page', () => {
   it('contains vocabulary title', () => {
-    cy.visit('/test/en') // go to the "Test ontology" home page
+    cy.visit('/yso/en') // Go to the "YSO - General Finnish ontology (archaeology)" home page
```
Comment (osma) [inline tests/cypress/template/vocab-home.cy.js:3]: The URL should be `/yso/en/` with the final slash included. (This was also wrong in the previous version)
The URL without the final slash will work, but there is an extra redirect because it's not the "canonical" URL of the vocabulary home page.

File: tests/cypress/template/vocab-home.cy.js [hunk 3]
```diff
@@ -1,42 +1,134 @@
...
 
-    // check that the vocabulary title is correct
-    cy.get('#vocab-title > a').invoke('text').should('equal', 'Test ontology')
+    // Check that the vocabulary title is not empty
+    cy.get('#vocab-title > a').invoke('text').should('equal', 'YSO - General Finnish ontology (archaeology)')
   })
   it('shows alphabetical index letters', () => {
-    cy.visit('/test/en') // go to the "Test ontology" home page
+    cy.visit('/yso/en') // Go to the "YSO - General Finnish ontology (archaeology)" home page
```
Comment (osma) [inline tests/cypress/template/vocab-home.cy.js:9]: Same URL issue here, and below a few more times

**Review by osma — CHANGES_REQUESTED** [review]
Looks good in general, I gave a few minor suggestions for improvements. After considering those, this can be merged.
